src/backend/distributed/utils/distribution_column.c

```diff
@@ -129,6 +132,12 @@ BuildDistributionKeyFromColumnName(Relation distributedRelation, char *columnNam
 	Var *distributionColumn = NULL;
 	char *tableName = RelationGetRelationName(distributedRelation);
 
+	/* short circuit for reference tables */
```

Comment:
I guess the alternative is to not call it for references tables, e.g.

```
    /* only build a distribution column for distributed tables, not reference tables */
    if (distributionMethod != DISTRIBUTE_BY_NONE)
    {
        distributionColumn = BuildDistributionKeyFromColumnName(relation,
                                                                distributionColumnName);
    }
```
in `create_distributed_table.c`, which would be clearer to the caller.